metpy/io/upperair.py

```diff
@@ -155,6 +163,16 @@ def get_data(time, site_id, region='naconf'):
                              'for station {stid}.'.format(time=time, region=region,
                                                           stid=site_id))
 
+        # See if the return is valid, but has no data
+        if data.find(b'too busy') != -1:
+            if retries > 0:
+                sleep(retry_delay / 1000.)
+                WyomingUpperAir.get_data(time, site_id, region, retries - 1)
```

Comment:
Good catches - I'm not clear on how to test it either. I've emailed Larry about getting a real return code.